File: app/src/main/java/au/id/blackwell/kurt/lantv/player/TvPlayerFactory.java
package au.id.blackwell.kurt.lantv.player;

import android.content.Context;

public class TvPlayerFactory {
    public static final String ANDROID = "android";
    public static final String IJK = "ijk";

    public TvPlayer create(String playerType, Context context) {
        switch (playerType) {
            case ANDROID:
                return new AndroidMediaPlayerView(context);
            case IJK:
                return new IjkMediaPlayerView(context);
            default:
                return null;
        }
    }
}

File: app/src/main/java/au/id/blackwell/kurt/lantv/player/TvPlayerFactory.kt
package au.id.blackwell.kurt.lantv.player

import android.content.Context

class TvPlayerFactory {

    fun create(playerType: String, context: Context): TvPlayer? {
        when (playerType) {
            ANDROID -> return AndroidMediaPlayerView(context)
            IJK -> return IjkMediaPlayerView(context)
            else -> return null
        }
    }

    companion object {
        val ANDROID = "android"
        val IJK = "ijk"
    }
}
